File: DAA/CPP/n-queens.cpp
#include <bits/stdc++.h>
using namespace std;

void printQueens(vector<int> Q) {
	for(int q: Q)
		cout << q+1 << " ";
	cout << "\n";
}

void placeQueens(vector<int> Q, int r) {
	if (r == Q.size())
		printQueens(Q);
	else {
		for(int j=0; j<Q.size(); j++) {
			bool legal=true;
			for(int i=0; i<r; i++) {
				if (Q[i] == j || Q[i] == j - r + i || Q[i] == j + r - i) {
					legal = false;
					break;
				}
			}
			if (legal) {
				Q[r] = j;
				placeQueens(Q, r+1);
			}
		}
	}
}

int main() {
	int n; cin >> n;
	vector<int> Q(n, 0);

	placeQueens(Q, 0);


	return 0;
}

File: DAA/n-queens.py

def placeQueens(Q, r):
  if r == len(Q):
    for x in Q:
      print(x+1, end = " ")
    print()
  else:
    for j in range(len(Q)):
      legal = True
      for i in range(r): 
        if (Q[i] == j or Q[i] == j - r + i or Q[i] == j + r - i):
          legal = False
          break
      if legal:
        Q[r] = j
        placeQueens(Q, r+1)

Q = [0] * 4
r = 0

placeQueens(Q, r)


      
